### PR TITLE
qt6Packages.qt6gtk2: init at 0.2

### LINKED ISSUE #244895: GTK/Qt theme not working
### Describe the bug
I've followed the instructions described in the [manual](https://nixos.org/manual/nixos/unstable/#sec-x11-gtk-and-qt-themes), but Qt applications are not using my GTK theme.

### Steps To Reproduce
```nix
qt.enable = true;
qt.platformTheme = "gtk2";
qt.style = "gtk2";
environment.systemPackages = [ pkgs.qalculate-qt ];
```

Running `qalculate-qt` will yield the following error:

```bash
QApplication: invalid style override 'gtk2' passed, ignoring it.
	Available styles: Windows, Fusion
```

### Expected behavior
Qt applications should properly detect the `gtk2` style.

### Notify maintainers

@romildo 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.39, NixOS, 23.11 (Tapir), 23.11.20230721.9ca7856`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - nixpkgs: `/nix/store/d1fspdh09fpk7dkl5s667vhpkgv12dsx-source`
```

### PR DESCRIPTION
**Pull request by Misterio77**
###### Description of changes

Add [qt6gtk2](https://github.com/trialuser02/qt6gtk2), plugins that help make qt6 apps use gtk2 theming.

I've also set qt's NixOS module to have qt6gtk2 available when using `platformTheme` gtk2.

Fixes https://github.com/NixOS/nixpkgs/issues/244895

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Misterio77** [comment]
Fixes https://github.com/NixOS/nixpkgs/issues/244895

**Comment by Misterio77** [comment]
While working on this, I noticed `/lib/qt-6/plugins` are only added to `environment.profileRelativeSessionVariables` when `qt.enable = true` (https://github.com/NixOS/nixpkgs/pull/211003).

I think this should always be the case; adding qt6 plugins to my (user) environment should make them work, similarly to [qt{4,5}](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/environment.nix#L41) and [GTK{2,3,4}](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/environment.nix#L43).

File: pkgs/tools/misc/qt6gtk2/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    cp -pr src/qt6gtk2-style/libqt6gtk2-style.so $out/lib/qt-6/plugins/styles
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "GTK+2.0 integration plugins for Qt6";
+    license = lib.licenses.gpl2Plus;
+    homepage = "https://github.com/trialuser02/qt6gtk2";
+    platforms = lib.platforms.linux;
```
Comment (thiagokokada) [inline pkgs/tools/misc/qt6gtk2/default.nix:33]: Can you add yourself as the maintainer?

Comment (Misterio77) [reply pkgs/tools/misc/qt6gtk2/default.nix:33]: Oopsie, forgot to! Addressed.

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**